pkg/cluster/awsiam.go

```diff
@@ -54,3 +59,12 @@ func processAWSIam(c *Config, objects ObjectLookup) {
 		}
 	}
 }
+
+func validateAWSIamConfigName(c *Config) error {
+	for _, idr := range c.Cluster.Spec.IdentityProviderRefs {
+		if idr.Kind == anywherev1.AWSIamConfigKind && c.AWSIAMConfigs == nil {
+			return fmt.Errorf("no %s named %s", anywherev1.AWSIamConfigKind, idr.Name)
```

Comment:
I would vote to have an error message a bit more descriptive, like:
```suggestion
			return fmt.Errorf("%s/%s referenced in Cluster but not present in Config", anywherev1.AWSIamConfigKind, idr.Name)
```

Or something else that gives us a bit more context on where the error is happening